### ISSUE TITLE
When trying to use `null` as a raw query binding, the values do not get bound, once the query is built it remains with the placeholder `?`

### ISSUE THREAD

**Issue by JulioPablo** [laravel/framework#35198]
- Laravel Version: 8
- PHP Version: 7.4
- Database Driver & Version: mysql - mariadb 10.2

### Description:

When trying to use `null` as a raw query binding, the values do not get bound, once the query is built it remains with the placeholder `?`

### Steps To Reproduce:

A query like
```
DB::statement("UPDATE my_table SET `column1` = ?, `column2` = ?", [1, null]);
```

produces 

```
UPDATE my_table SET `column1` = 1, `column2` = ?;
```
Which is, of course, invalid SQL.

**Comment by driesvints** [comment]
Not so sure what you expected to happen here tbh. it's not like the query will be magically updated to `IS NULL`.

**Comment by JulioPablo** [comment]
Why would it update to ```IS NULL```?

It should set the value of column2 to null, the correct query would be 

```
UPDATE my_table SET `column1` = 1, `column2` = null;
```
Maybe try reading the issue a with some more care before just closing it?

**Comment by driesvints** [comment]
Heya you're correct. We're checking in on this 👍

**Comment by driesvints** [comment]
@JulioPablo I can't reproduce this on MySQL 8. Can you perhaps try MySQL as well?

**Comment by JulioPablo** [comment]
@driesvints At this very moment no, but maybe during the weekend I can set a docker container with MySQL and report back.

**Comment by driesvints** [comment]
@JulioPablo feel free to report back when you're able to 👍

**Comment by JulioPablo** [comment]
Does laravel only support MySQL, I mean, if the issue is only present in MariaDB 10.2 onwards, is it not a issue?

**Comment by driesvints** [comment]
In general we do try to support MariaDB as much as we possible can (although it's not listed as a supported engine: https://laravel.com/docs/8.x/database#introduction). But let's first confirm if this does or doesn't work for you on MySQL either.

**Comment by JulioPablo** [comment]
Gotcha.

**Comment by rodrigopedra** [comment]
Just tested with MySQL 8 and it is working.

- PHP version: 7.4.12
- Laravel version: 8.14.0
- MySQL version: 8.0.22

Test code:

~~~php
<?php

use Illuminate\Support\Facades\DB;
use Illuminate\Support\Facades\Route;

Route::get('/', function () {
    DB::statement('UPDATE `my_table` SET `column1` = ?, `column2` = ?', [1, null]);

    return DB::table('my_table')->get();
});
~~~

Before running I created a table with two nullable integer columns named after this test code and added some rows there.

Table before running test route:

| column1 | column2 |
| :--- | :--- |
| 1 | 3 |
| 2 | 4 |

Table after running test route:

| column1 | column2 |
| :--- | :--- |
| 1 | NULL |
| 1 | NULL |

Testing route output:

~~~json
[{"column1":1,"column2":null},{"column1":1,"column2":null}]
~~~

Hope it helps.

**Comment by driesvints** [comment]
@rodrigopedra are you in any position to also test this with MariaDB maybe? Thanks for helping :)

**Comment by rodrigopedra** [comment]
Just installed, and it works as expected.

MariaDB version: 10.5.8-MariaDB

As I installed MariaDB alongside my local MySQL installation (I used this guide: https://mariadb.com/kb/en/installing-mariadb-alongside-mysql/ including some of the comments down below it), I first had some problems connecting to MariaDB from Laravel, as I configured it to use a different socket and port than MySQL's.

So it was giving this error:

~~~
SQLSTATE[HY000] [1045] Access denied for user 'root'@'localhost' (using password: YES) (SQL: UPDATE `my_table` SET `column1` = 1, `column2` = ?) 
~~~

Where it appears there is a parsing error, but it is actually a username/password error.

After fixing the socket configuration (setting `DB_SOCKET`  in my `.env` file), I was able to run the same tests I ran in my last response.

The reason I think the error message leads to one think it is a parsing error at first sight, is that (I'm guessing) when PHP's PDO is binding the parameters, it makes a callto the database to sanitize values it doesn't know how to properly bind. As first variable is a integer PDO does by itself, but when binding `NULL` it might be trying to check with the database for datatype, etc. As I said earlier, I am just guessing. But that would explain why the binding is partially done.

Turns out the error message is very easy to reproduce, even with MySQL. Just use a wrong username or password, and the error will show a partially bound query.

Maybe @JulioPablo could post the whole error message, and confirm it is not a database credentials/access problem on his installation.

Hope it helps.
